certification/internal/shell/has_v2_related_image_schema_version.go

```diff
@@ -0,0 +1,229 @@
+package shell
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/api/pkg/manifests"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
+	log "github.com/sirupsen/logrus"
+)
+
+// RelatedImagesAreSchemaVersion2Check is part of the Operator policy and implements
+// the Check interface.
+type RelatedImagesAreSchemaVersion2Check struct{}
+
+// Validate checks the image manifest for each related image referenced in a
+// ClusterServiceVersion and ensures that the schema version used is version 2.
+func (p *RelatedImagesAreSchemaVersion2Check) Validate(image string) (bool, error) {
+
+	imageToSchemaVersion, err := p.getDataToValidate(image)
+	if err != nil {
+		return false, fmt.Errorf("%w: %s", errors.ErrRunContainerFailed, err)
+	}
+
+	return p.validate(imageToSchemaVersion)
+}
+
+// getDataToValidate pulls a ClusterServiceVersion from the input operator bundle,
+// checks the ClusterServiceVersion's related images declaration, and assembles a list of
+// images and their image manifest schema version values.
+func (p *RelatedImagesAreSchemaVersion2Check) getDataToValidate(image string) (map[string]int, error) {
+	bundlePath, err := p.extractBundleFromImage(image)
+	if err != nil {
+		log.Error(err)
+		return nil, fmt.Errorf("%w: %s", errors.ErrDeterminingRelatedImageSchemaVers, err)
+	}
+	csv, err := p.readBundle(bundlePath)
+	if err != nil {
+		log.Error(err)
+		return nil, fmt.Errorf("%w: %s", errors.ErrDeterminingRelatedImageSchemaVers, err)
+	}
+
+	relatedImages, err := p.getRelatedImagesForCSV(csv)
+	if err != nil {
+		log.Error(err)
+		return nil, fmt.Errorf("%w: %s", errors.ErrDeterminingRelatedImageSchemaVers, err)
+	}
+
+	imageSchemaVersionMap, err := p.inspectSchemaVersionForImage(relatedImages)
+	if err != nil {
+		log.Error(err)
+		return nil, fmt.Errorf("%w: %s", errors.ErrDeterminingRelatedImageSchemaVers, err)
+	}
+
+	return imageSchemaVersionMap, nil
+}
+
+// extractBundleFromImage will extract the bundle contents from the specified image, and return the
+// path to the bundle on the filesystem. This is done by creating the container, copying the contents
+// out of the container, and then cleaning up the container.
+func (p *RelatedImagesAreSchemaVersion2Check) extractBundleFromImage(image string) (string, error) {
+	localBundlePath := "extracted-bundle"
+
+	// create the container. Bundle containers are non-runnable, so we need to
+	// add a dummy entrypoint so that we can create the container.
+	createReport, err := podmanEngine.Create(image, &cli.PodmanCreateOptions{Entrypoint: "false"})
```

Comment:
As an alternative to running a stopped container would the following work?
`podman image mount registry.redhat.io/quay/quay-operator-bundle` for example.
then pass `<value from last command>/manifests/` to readBundle(...

Reply:
@tonytcampbell I wish! Unfortunately, it doesn't play out that way - 

```
(nonroot-user)$ podman image mount registry.redhat.io/quay/quay-operator-bundle
Error: cannot run command "podman image mount" in rootless mode, must execute `podman unshare` first
```

I discussed this with @bcrochet recently, and this is what led to the implementation submitted in #126. 

#126 solves this by executing a single check of preflight in an rootless mode obtained exactly how that error text states.

This PR does utilize the alternate approach (create a stopped container and pulling data from it).